fs/fcb/src/fcb.c

```diff
@@ -223,18 +224,17 @@ fcb_offset_last_n(struct fcb *fcb, uint8_t entries, uint32_t *last_n_off)
     while (!fcb_getnext(fcb, &loc)) {
         if (i == 0) {
             /* Start from the beginning of fcb entries */
-            *last_n_off = loc.fe_elem_off;
-            start = loc;
+            *last_n_entry = start = loc;
         }
-        /* Update last_n_off after n entries and keep updating */
-        if (i >= (entries - 1)) {
+        /* Update last_n_entry after n entries and keep updating */
+        else if (i > entries) {
             fcb_getnext(fcb, &start);
-            *last_n_off = start.fe_elem_off;
+            *last_n_entry = start;
         }
         i++;
     }
 
-    return 0;
+    return (entries + 1) > i ? OS_ENOENT : 0;
```

Comment:
This should return 0 even if fcb has less than number of entries asked. User of this interface is trying to copy up to 'entries' number of fcb elements.